extensions/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/supervisor/KafkaSupervisor.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.indexing.kafka.supervisor;
+
+import com.google.common.base.Optional;
+import com.google.inject.Inject;
+import com.google.inject.Provider;
+import com.metamx.common.concurrent.ScheduledExecutorFactory;
+import com.metamx.common.concurrent.ScheduledExecutors;
+import com.metamx.common.lifecycle.LifecycleStart;
+import com.metamx.common.lifecycle.LifecycleStop;
+import com.metamx.emitter.EmittingLogger;
+import io.druid.indexing.overlord.supervisor.IngestionSupervisor;
+import io.druid.indexing.overlord.supervisor.SupervisorWorker;
+import org.joda.time.Duration;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ScheduledExecutorService;
+
+/**
+ * Manages the creation and execution of {@link KafkaSupervisorWorker}.
+ */
+public class KafkaSupervisor implements IngestionSupervisor<KafkaSupervisorJob>
```

Comment:
This doesn't seem to be doing really Kafka-specifc stuff… it seems like it'd make more sense to:

- Move this class to core and call it "SupervisorManager"
- Rename "Worker" to "Supervisor" (worker already means something else in the indexing service anyway)

The advantage of that being that modules that want to implement supervision don't need to have a class like this one, as the SupervisorManager in core would handle management of all of them.